Signum.Engine/Operations/OperationLogic.cs

```diff
@@ -105,9 +106,55 @@ public static void Start(SchemaBuilder sb)
             sb.Schema.SchemaCompleted += () => RegisterCurrentLogs(sb.Schema);
 
             ExceptionLogic.DeleteLogs += ExceptionLogic_DeleteLogs;
+
+            OperationsToken.GetTypeEligibleOperations = (entityType) => OperationsToken_GetTypeEligibleOperations(entityType);
+            OperationToken.IsAllowedExtension = (operationSymbol, entityType) => OperationToken_IsAllowedExtension(operationSymbol, entityType);
+            OperationToken.BuildExtension = (entityType, operationSymbol, parentExpression) => OperationToken_BuildExpression(entityType, operationSymbol, parentExpression);
         }
     }
 
+    private static IEnumerable<OperationSymbol> OperationsToken_GetTypeEligibleOperations(Type entityType)
+    {
+        return TypeOperations(entityType)
+            .Where((o) =>
+                {
+                    var op = TryFindOperation(entityType, o.OperationSymbol) as IEntityOperation;
+                    return op?.CanExecuteExpression() != null || op?.HasCanExecute == false;
+                })
+            .Select(o => o.OperationSymbol);
+    }
+
+    private static Expression OperationToken_BuildExpression(Type entityType, OperationSymbol operationSymbol, Expression parentExpression)
+    {
+        var operation = (IEntityOperation)FindOperation(entityType, operationSymbol);
+
+        LambdaExpression? cee = operation.CanExecuteExpression();
+
+        if (cee == null && operation.HasCanExecute)
+                throw new InvalidOperationException(OperationMessage.Operation01DoesNotHaveCanExecuteExpression
+                                                                    .NiceToString().FormatWith(operationSymbol.NiceToString(), operationSymbol.Key));
+              
+        var entity = parentExpression.ExtractEntity(false);
+        var operationKey = Expression.Constant(operationSymbol.Key);
+        var canExecute = cee == null ? Expression.Constant(null, typeof(string)) : 
+                ExpressionReplacer.Replace(cee.Body, new Dictionary<ParameterExpression, Expression> { { cee.Parameters.Single(), entity } });
+
+        var dtoConstructor = typeof(CellOperationDTO).GetConstructor(new[] { typeof(Lite<IEntity>), typeof(string),  typeof(string) });
+
+        NewExpression newExpr = Expression.New(dtoConstructor!, entity.BuildLite(), operationKey, canExecute);
+
+        return newExpr;
+    }
+
+    private static string? OperationToken_IsAllowedExtension(OperationSymbol operationSymbol, Type entityType)
+    {
+        var operation = (IEntityOperation)FindOperation(entityType, operationSymbol);
+        if (operation.CanExecuteExpression() == null && operation.HasCanExecute)

```

Comment:
Necessary? Isn’t already filtered above by having CanExecute?

Reply:
I had kept it just for some guaranty 😉, but you're quite right it never helps anymore
removed.